hibernate-reactive-core/src/main/java/org/hibernate/reactive/query/sqm/mutation/internal/temptable/ReactiveGlobalTemporaryTableStrategy.java

```diff
@@ -65,22 +65,23 @@ default void prepare(MappingModelCreationProcess mappingModelCreationProcess, Jd
 		if ( !createIdTables ) {
 			tableCreatedStage.complete( null );
 		}
-
-		LOG.debugf( "Creating global-temp ID table : %s", getTemporaryTable().getTableExpression() );
-
-		connectionStage()
-				.thenCompose( this::createTable )
-				.whenComplete( (connection, throwable) -> releaseConnection( connection )
-						.thenAccept( v -> {
-							if ( throwable == null ) {
-								setDropIdTables( configService );
-								tableCreatedStage.complete( null );
-							}
-							else {
-								tableCreatedStage.completeExceptionally( throwable );
-							}
-						} )
-				);
+		else {
+			LOG.debugf( "Creating global-temp ID table : %s", getTemporaryTable().getTableExpression() );
+
+			connectionStage()
+					.thenCompose( this::createTable )
+					.whenComplete( (connection, throwable) -> releaseConnection( connection )
+							.thenAccept( v -> {
+								if ( throwable == null ) {
+									setDropIdTables( configService );
+									tableCreatedStage.complete( null );
+								}
+								else {
+									tableCreatedStage.completeExceptionally( throwable );
+								}
+							} )
+					);
+		}
```

Comment:
This solution is fine, but we can keep the changes smaller and the code more similar to ORM if we add the `return` in the if-block: 
```
if ( !createIdTables ) {
   tableCreatedStage.complete( null );
   return;
}
```

It also reduces the indentation.

Reply:
Sure... I'm fine with the original solution